test/Test/Prolens/Property.hs

```diff
@@ -61,6 +61,20 @@ profunctorsSpec = describe "Profunctor" $ do
             eqFun
                 (dimap (ab . bc) (yz . xy) f)
                 (dimap bc yz $ dimap ab xy f)
+    describe "Forget" $ do
+        it "Identity: dimap id id ≡ id" $ hedgehog $ do
+            f <- unForget <$> forAllWith (const "f") genForget
+            x <- forAll genInt
+            dimap id id f x === f x
+        it "Composition: dimap (ab . bc) (yz . xy) ≡ dimap bc yz . dimap ab xy" $ hedgehog $ do
+            f  <- unForget <$> forAllWith (const "f")  genForget
+            ab <- unForget <$> forAllWith (const "ab") genForget
+            bc <- unForget <$> forAllWith (const "bc") genForget
+            xy <- unForget <$> forAllWith (const "xy") genForget
+            yz <- unForget <$> forAllWith (const "xy") genForget
+
+            n <- forAll genInt
+            dimap (ab . bc) (yz . xy) f n === (dimap bc yz . dimap ab xy) f n
```

Comment:
It looks like this property doesn't test the instance for `Forget`, but it still tests the instance for function. Your property generates `Forget` but then unwraps it and gets an ordinary function. So, here actually the `Profunctor` instance for `(->)` is tested.

I think you need to write a function `eqForget` similar to `eqFun` below, unwrap `Forget` inside that function and use it here.

Let us know if you need any help with this :slightly_smiling_face:

Reply:
Thanks @chshersh I knew I was missing something! Thank you for the guidance. Please, let me know if the property tests look better now.